### ISSUE TITLE
Main EditorConfig vs. "Working" EditorConfig

### ISSUE THREAD

**Issue by brock** [editorconfig/editorconfig#201]
From what I understand about EditorConfig after using it for a bit is that it forces the codebase and all working environments to use the same coding style. That's great.

What I'm interested in exploring is the possibility of creating a "working" editorconfig. And by that I mean, a user can make their editor look however they want, but git commits adhere to the "main" editorconfig file, or the one found up the tree.

I imagine a grunt task or a pre-commit hook could accomplish what I'm after. This might go against this project's goals, but just wanted to ask here if anyone is already doing something similar, or has tried this and can share issues they faced.

**Comment by xuhdev** [comment]
> a user can make their editor look however they want

What does your "look" mean? Does it mean any individual's editor just has its own "looking", or the files are also personalized?

**Comment by brock** [comment]
@xuhdev that's right - by "look" I just meant that maybe I prefer to use 4 spaces, but the project uses 2 spaces. In that case, I'd want to work using 4 spaces, but make commits using 2 spaces.

**Comment by jedmao** [comment]
@brock I've been dreaming about that type of feature forever, Man. We're on the same page. I think the logistics of implementing such a feature would be a nightmare, but I would definitely support the endeavor. It's a slick feature. Basically, it gives you the luxury of setting your tab_width in a spaces world. Love it! :star2:

**Comment by treyhunner** [comment]
@brock this is not a solvable problem in the general case without some very intelligent handling of a number of common edge cases.

The main problem is alignment vs. indentation.  @jedmao wrote a blog post that discusses this problem briefly: http://jedmao.ghost.io/2014/08/20/tabs-vs-spaces-the-age-old-war/

**Comment by brock** [comment]
Thanks @treyhunner. That was a good read. I imagine something like this would be best if it could somehow be managed entirely by the user, without ever having to involve the project. If I'm able to daydream a bit, imagine I can set my preferences at my $HOME level. Then any projects would honor my preferred style, and format to any defined editorConfig settings if they exist. I'd be the only one who would ever know that I'm operating differently from the team, and I'd be responsible for verifying that this conversion operation was successful.

**Comment by jedmao** [comment]
@treyhunner is right. The reason this is not a solvable problem is because you can't reverse-engineer spaces for indentation and interpret what was intended for indentation vs. alignment. Not without a concrete syntax tree, of which there is none.

**Comment by cxw42** [comment]
I am closing this issue as beyond the scope of editorconfig, since it would indeed require language-specific tooling as noted above.
